bodhi/tests/server/consumers/test_greenwave.py

```diff
@@ -0,0 +1,104 @@
+                'crypto': 'x509',
+                'topic': 'org.fedoraproject.prod.greenwave.decision.update',
+                'signature': '100% real please trust me',
+                'msg': {
+                    'subject_type': 'koji_build',
+                    'policies_satisfied': True,
+                    'decision_context': 'bodhi_update_push_testing',
+                    'satisfied_requirements': [
+                        {
+                            'testcase': 'dist.abicheck',
+                            'type': 'test-result-passed',
+                            'result_id': 27968725
+                        },
+                        {
+                            'testcase': 'dist.rpmdeplint',
+                            'type': 'test-result-passed',
+                            'result_id': 28670567
+                        }
+                    ],
+                    'product_version': 'fedora-30',
+                    'applicable_policies': [
+                        'taskotron_release_critical_tasks_for_testing'
+                    ],
+                    'unsatisfied_requirements': [],
+                    'subject_identifier': 'bodhi-3.13.3-1.fc30',
+                    'subject': [
+                        {
+                            'item': 'bodhi-3.13.3-1.fc30',
+                            'type': 'koji_build'
+                        }
+                    ],
+                    'summary': 'All required tests passed',
+                    'previous': {
+                        'satisfied_requirements': [
+                            {
+                                'testcase': 'dist.abicheck',
+                                'type': 'test-result-passed',
+                                'result_id': 27968725
+                            },
+                            {
+                                'testcase': 'dist.rpmdeplint',
+                                'type': 'test-result-passed',
+                                'result_id': 28666088
+                            }
+                        ],
+                        'summary': 'All required tests passed',
+                        'policies_satisfied': True,
+                        'unsatisfied_requirements': [],
+                        'applicable_policies': [
+                            'taskotron_release_critical_tasks_for_testing'
+                        ],
+                    }
+                }
+            },
+        )
+        self.handler = greenwave.GreenwaveHandler()
+
+    @mock.patch('bodhi.server.consumers.greenwave.Update')
+    def test_update_test_gating_status(self, mock_build_model):
+        """Assert that messages marking the build as signed updates the database"""
+        update = mock_build_model.get.return_value.update
+        update.test_gating_status = models.TestGatingStatus.waiting
+
+        self.handler(self.sample_message)
+        self.assertTrue(update.test_gating_passed)
```

Comment:
Is this test making a live web request to Greenwave? If so, that will not pass in CI (CI disables networking to ensure that the unit tests do not use network resources.)

Reply:
I believe it is.
I'm currently researching how to mock that.

Reply:
The live request to Greenwave was made during 
`update.update_test_gating_status()`
I have mock'ed that, and it is no longer doing a live contact to greenwave during the tests.  and the test runs much faster now as well.